src/support.cc

```diff
@@ -32,6 +32,18 @@
 #include <sys/socket.h>
 #include <string>
 
+#define EXT_STRUCT_VERSION  "1.3.6.1.4.1.3704.1.1"
```

Comment:
Looking at the usage of these mnemonics, they seem to be only used in the `else` block in `src/support.cc`, here:

```
2673   if (!vcek) {
2674     add_ext(x509, NID_key_usage, "critical,keyCertSign,digitalSignature,cRLSign");
2675     add_ext(x509,  NID_ext_key_usage, "clientAuth,serverAuth");
2676     // add_ext(x509, NID_subject_key_identifier, "hash");
2677     if (is_root) {
2678       add_ext(x509, NID_basic_constraints, "critical,CA:TRUE");
2679     }
2680   } else {
2681     // Add custom extensions if we are generating VCEK for the SEV-SNP simulator
```

So if these constants are all VCEK-related, how about naming them all with `VCEK_` prefix, like `VCEK_STRUCT_VERSION`, `VCEK_BLSPL` and so on.

`EXT` just means external, but is not indicative that these mnemonics are really only related to VCEK-stuff.

If you do want to retain `EXT` also, the preferred name would be `VCEK_EXT_STRUCT_VERSION` and so on. A bit longer ... but less ambiguous.

Reply:
Also, over time, the provenance of these constants will be lost. 

If they are coming from some external AMD-VCEK spec, can you xref its url here?

If they are simply some random constants chosen to exercise the test-case & simulator, can you please add a comment to that effect. Something on the lines of:

"Constants are chosen arbitrarily just to have some data for the custom extensions to check-against in the simulated VCEK-certificates we internally generate for testing."

Reply:
These are OIDs applied by and belong to AMD. So, I will add the xref of the spec. Will also update the macro names.